### ISSUE TITLE
install on windows

### ISSUE THREAD

**Issue by curlysasha** [chaojie/ComfyUI-Moore-AnimateAnyone#1]
followed your instructions, but
![Screenshot_1](https://github.com/chaojie/ComfyUI-Moore-AnimateAnyone/assets/116119767/fb8e5d63-fd1a-4962-a25e-a77990d1a5d5)

**Comment by Kobak24** [comment]
got the same error:
  File "E:\_AI\ComfyUI\nodes.py", line 1872, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "E:\_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\__init__.py", line 1, in <module>
    from .nodes import NODE_CLASS_MAPPINGS
  File "E:\_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\nodes.py", line 22, in <module>
    from src.models.unet_2d_condition import UNet2DConditionModel
  File "E:\_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\src\models\unet_2d_condition.py", line 40, in <module>
    from .unet_2d_blocks import (
  File "E:\_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\src\models\unet_2d_blocks.py", line 15, in <module>
    from .transformer_2d import Transformer2DModel
  File "E:\_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\src\models\transformer_2d.py", line 7, in <module>
    from diffusers.models.embeddings import CaptionProjection
ImportError: cannot import name 'CaptionProjection' from 'diffusers.models.embeddings' (C:\Users\****\AppData\Local\Programs\Python\Python311\Lib\site-packages\diffusers\models\embeddings.py)

Cannot import E:\_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone module for custom nodes: cannot import name 'CaptionProjection' from 'diffusers.models.embeddings' (C:\Users\****\AppData\Local\Programs\Python\Python311\Lib\site-packages\diffusers\models\embeddings.py)

I got this version of diffusers: 
Name: diffusers
Version: 0.25.0

Have you tried the Tri fix button?

**Comment by curlysasha** [comment]
> got the same error: File "E:_AI\ComfyUI\nodes.py", line 1872, in load_custom_node module_spec.loader.exec_module(module) File "", line 940, in exec_module File "", line 241, in _call_with_frames_removed File "E:_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone__init__.py", line 1, in from .nodes import NODE_CLASS_MAPPINGS File "E:_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\nodes.py", line 22, in from src.models.unet_2d_condition import UNet2DConditionModel File "E:_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\src\models\unet_2d_condition.py", line 40, in from .unet_2d_blocks import ( File "E:_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\src\models\unet_2d_blocks.py", line 15, in from .transformer_2d import Transformer2DModel File "E:_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\src\models\transformer_2d.py", line 7, in from diffusers.models.embeddings import CaptionProjection ImportError: cannot import name 'CaptionProjection' from 'diffusers.models.embeddings' (C:\Users****\AppData\Local\Programs\Python\Python311\Lib\site-packages\diffusers\models\embeddings.py)
> 
> Cannot import E:_AI\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone module for custom nodes: cannot import name 'CaptionProjection' from 'diffusers.models.embeddings' (C:\Users****\AppData\Local\Programs\Python\Python311\Lib\site-packages\diffusers\models\embeddings.py)
> 
> I got this version of diffusers: Name: diffusers Version: 0.25.0
> 
> Have you tried the Tri fix button?

yeah, tryied all buttons )
i think requirements need to be installed in venv, but i dont know how to do it in comfy

**Comment by curlysasha** [comment]
install via git link in manager
then python tools/download_weights.py 
then restart server

working

**Comment by mrkennypowers** [comment]
> install via git link in manager then python tools/download_weights.py then restart server
> 
> working

I wish this were true, but I've tried installing this way and every other way i can think of besides a brand new comfy ui installation and each time the nodes fail to load.

**Comment by curlysasha** [comment]
> > install via git link in manager then python tools/download_weights.py then restart server
> > working
> 
> I wish this were true, but I've tried installing this way and every other way i can think of besides a brand new comfy ui installation and each time the nodes fail to load.

delete this node folder and try again

**Comment by mrkennypowers** [comment]
> > > install via git link in manager then python tools/download_weights.py then restart server
> > > working
> > 
> > 
> > I wish this were true, but I've tried installing this way and every other way i can think of besides a brand new comfy ui installation and each time the nodes fail to load.
> 
> delete this node folder and try again

Yup, after moving the pretrained weights that worked. Ty

**Comment by konradre** [comment]
> > > > install via git link in manager then python tools/download_weights.py then restart server
> > > > working
> > > 
> > > 
> > > I wish this were true, but I've tried installing this way and every other way i can think of besides a brand new comfy ui installation and each time the nodes fail to load.
> > 
> > 
> > delete this node folder and try again
> 
> Yup, after moving the pretrained weights that worked. Ty

Where did you move the weights? I think what happens is if you downloaded them once theyre just simlinks to the cache, so have to delete cache before attempting again. But you resolved it by moving the weights folder?

**Comment by curlysasha** [comment]
> > > > > install via git link in manager then python tools/download_weights.py then restart server
> > > > > working
> > > > 
> > > > 
> > > > I wish this were true, but I've tried installing this way and every other way i can think of besides a brand new comfy ui installation and each time the nodes fail to load.
> > > 
> > > 
> > > delete this node folder and try again
> > 
> > 
> > Yup, after moving the pretrained weights that worked. Ty
> 
> Where did you move the weights? I think what happens is if you downloaded them once theyre just simlinks to the cache on subsequent installs, so have to delete cache before attempting again. But you resolved it by moving the weights folder?

Simlink it ok

**Comment by konradre** [comment]
> > > > > > install via git link in manager then python tools/download_weights.py then restart server
> > > > > > working
> > > > > 
> > > > > 
> > > > > I wish this were true, but I've tried installing this way and every other way i can think of besides a brand new comfy ui installation and each time the nodes fail to load.
> > > > 
> > > > 
> > > > delete this node folder and try again
> > > 
> > > 
> > > Yup, after moving the pretrained weights that worked. Ty
> > 
> > 
> > Where did you move the weights? I think what happens is if you downloaded them once theyre just simlinks to the cache on subsequent installs, so have to delete cache before attempting again. But you resolved it by moving the weights folder?
> 
> Simlink it ok



> > > > > > install via git link in manager then python tools/download_weights.py then restart server
> > > > > > working
> > > > > 
> > > > > 
> > > > > I wish this were true, but I've tried installing this way and every other way i can think of besides a brand new comfy ui installation and each time the nodes fail to load.
> > > > 
> > > > 
> > > > delete this node folder and try again
> > > 
> > > 
> > > Yup, after moving the pretrained weights that worked. Ty
> > 
> > 
> > Where did you move the weights? I think what happens is if you downloaded them once theyre just simlinks to the cache on subsequent installs, so have to delete cache before attempting again. But you resolved it by moving the weights folder?
> 
> Simlink it ok

The fix doesnt work for me, no matter how many times I try it.

**Comment by Kobak24** [comment]
> > > > install via git link in manager then python tools/download_weights.py then restart server
> > > > working
> > > 
> > > 
> > > I wish this were true, but I've tried installing this way and every other way i can think of besides a brand new comfy ui installation and each time the nodes fail to load.
> > 
> > 
> > delete this node folder and try again
> 
> Yup, after moving the pretrained weights that worked. Ty

have you moved the pretrained weights somewhere? or you meant that you've removed and redownloaded them?

**Comment by paulperrone** [comment]
This worked for me, hope it helps.

TL;DR: On Windows use the python_embeded executable from the command line.

Prerequisite: Download ComfyUI-Moore-AnimateAnyone via the ComfyUI Manager or GitHub

1. Open up a new command line window.

2. CD to the root directory of your ComfyUI installation

3. Run the following command in the command line ...

```python_embeded\python.exe -m pip install -r .\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\requirements.txt```

4. Download the weights.

```python .\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\tools\download_weights.py```

5. Move the pretrained_weights dir from the root ComfyUI directory into the Moore-AnimateAnyone dir

YOUR-COMFY-ROOT\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\pretrained_weights

**Comment by konradre** [comment]
> This worked for me, hope it helps.
> 
> TL;DR: On Windows use the python_embeded executable from the command line.
> 
> Prerequisite: Download ComfyUI-Moore-AnimateAnyone via the ComfyUI Manager or GitHub
> 
> 1. Open up a new command line window.
> 2. CD to the root directory of your ComfyUI installation
> 3. Run the following command in the command line ...
> 
> `python_embeded\python.exe -m pip install -r .\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\requirements.txt`
> 
> 4. Download the weights.
> 
> `python .\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\tools\download_weights.py`
> 
> 5. Move the pretrained_weights dir from the root ComfyUI directory into the Moore-AnimateAnyone dir
> 
> YOUR-COMFY-ROOT\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\pretrained_weights

You sir, are legendary

**Comment by cardenluo** [comment]
![image](https://github.com/chaojie/ComfyUI-Moore-AnimateAnyone/assets/133877069/928428a9-7433-4802-95ab-4d72f73e2405)

**Comment by konradre** [comment]
Actually, moving the folder didnt seem to work for me, but installing the custom node from root folder fixed the 'Failed to Import' issue. But when moving pretrained weights folder from root after downloading weights, I got errors when trying to Queue Prompt. The solution in my case was to delete the weights folder again and run the .py file from the custom node AnimateAnyone folder.

**Comment by Ryugon07** [comment]
> Actually, moving the folder didnt seem to work for me, but installing the node through comfyUI manager, and the requirements.txt from comfyUI root folder fixed the 'Failed to Import' issue. But when moving pretrained weights folder from root after downloading weights, I got errors when trying to Queue Prompt. The solution in my case was to delete the weights folder again and run the .py file from the custom node AnimateAnyone folder.
> 
> To reiterate, the only solution I found was:
> 
> Download ComfyUI-Moore-AnimateAnyone via the ComfyUI Manager
> 
> Open up a new command line window.
> 
> CD to the root directory of your ComfyUI installation
> 
> Run the following command in the command line ...
> 
> `python_embeded\python.exe -m pip install -r .\ComfyUI\custom_nodes\ComfyUI-Moore-AnimateAnyone\requirements.txt`
> 
> CD to the ComfiUI-Moore-AnimateAnyone folder in your custom_nodes folder
> 
> Download the weights `python tools\download_weights.py`

Thanks Sire!

**Comment by howitzer99** [comment]
I've followed the steps above, and it's not working for me.  I keep getting the 'Failed to Import' error in the startup console, and the ComfyUI Manager still shows (IMPORT FAILED) in the Install Missing Plugins window.

**Comment by ctankep** [comment]
The `download_weights.py` is using `hf_hub_download` helper.
So it creates a hidden `.cache/huggingface` folder in your root then attempts to symlink back to it. Then the paths the repo is using to look for `pretrained_weights` on start up includes linux forward slashes so thats why that fails too. This could be made a lot simpler by just listing which models to use from " animate anyone " and let users set it up without issues ..
